controller/linodeobjectstoragebucket_controller.go

```diff
@@ -191,11 +204,47 @@ func (r *LinodeObjectStorageBucketReconciler) reconcileCreate(ctx context.Contex
 }
 
 func (r *LinodeObjectStorageBucketReconciler) reconcileUpdate(ctx context.Context, logger logr.Logger, bucketScope *scope.ObjectStorageBucketScope) error {
-	panic("unimplemented")
+	return nil
 }
 
 func (r *LinodeObjectStorageBucketReconciler) reconcileDelete(ctx context.Context, logger logr.Logger, bucketScope *scope.ObjectStorageBucketScope) error {
-	panic("unimplemented")
+	name := types.NamespacedName{Namespace: bucketScope.Object.Namespace, Name: bucketScope.Object.Name}
+	bucket := &infrav1alpha1.LinodeObjectStorageBucket{}
+	if err := r.Client.Get(ctx, name, bucket); err != nil {
+		if apierrors.IsNotFound(err); err != nil {
+			logger.Error(err, "Failed to fetch bucket")
+			return err
+		}
+		return nil
+	}
+
+	// Delete the OBJ bucket.
+	if err := services.DeleteObjectStorageBucket(ctx, bucketScope, logger); err != nil {
+		return fmt.Errorf("delete object storage bucket: %w", err)
+	}
+
+	// Remove the finalizer.
+	// This will allow the ObjectStorageBucket object to be deleted.
+	if err := bucketScope.RemoveFinalizer(ctx); err != nil {
+		return nil
+	}
+
+	return nil
+}
+
+// bucketExists indicates whether or not an Object Storage Bucket exists.
+// It uses the Linode API.
+func (r *LinodeObjectStorageBucketReconciler) bucketExists(ctx context.Context, logger logr.Logger, bucketScope *scope.ObjectStorageBucketScope) (bool, error) {
```

Comment:
I think this function should be in the `cloud/services` package since all Linode client code lives there.

Reply:
Agreed. Moved it